src/python/grpcio_testing/grpc_testing/__init__.py

```diff
@@ -0,0 +1,441 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fixtures for use in testing gRPC Python-using application code."""
+
+import abc
+import collections
+
+import grpc
+import six
+
+
+class ChannelRpc(six.with_metaclass(abc.ABCMeta)):
```

Comment:
Mmm, carnality... wait, we can't talk about that in the workplace! :-P

Do you mean one class for each of the four cardinalities, for a total of four classes? `UnaryUnaryChannelRpc`, `UnaryStreamChannelRpc`, and so forth? I agree that each would present a better, safer, and more helpful API than that presented by the one all-cardinalities `ChannelRpc` right now.

Is there something that would allow two classes that I'm not seeing?